content/en/observability_pipelines/performance.md

```diff
@@ -0,0 +1,24 @@
+- Observability Pipelines propagates backpressure, which is a signal that events cannot be processed as soon as they are received.
+- Observability Pipelines components also have in-memory buffering in case the component receiving the data is busy.
+
+## Backpressure
+
+Backpressure is a signal that events cannot be processed as soon as they are received, and that signal is propagated up through the pipeline. For example, if a processor is sending events to a sink, and the sink cannot process it as fast as it is receiving the events, backpressure is propagated from the sink to the processor and all the way back to the source.
+
+Backpressure determines if the system should slow down the consumption or acceptance of events because it is too busy to handle more work. In some cases though, the system should not immediately propagate backpressure, because this could lead to the constant slowing down of upstream components, and potentially cause issues outside of Observability Pipelines. For example, the system avoids slowing processes down when a component just crosses over the threshold of being fully saturated. The system is also able to handle temporary slowdowns and outages with external services that destinations send data to.
+
+## In-memory buffering for components
+
+All components in Observability Pipelines have a small in-memory buffer between them. The buffer is the channel that two components communicate over. It ensures that there is a small amount of space, typically 100 events, that can be used to send events even if the component on the receiving end is busy. This allows maximizing throughput when workloads are not entirely uniform.
+
+Buffering protects against temporary overloads or outages for a given workload. The buffering model prioritizes performance when handling an excess of events, an amount that is beyond what a destination can process, by using in-memory buffers on destinations. By default, a destination's default buffer size is increased from 100 events to 500 events. The buffer capacity is increased because destinations are typically the primary source of backpressure in any given Observability Pipelines topology. They communicate to services over the network, where latency may be introduced or outages may temporarily occur.
+
+Observability Pipelines destination's buffers are configured to block events, which means it waits indefinitely to write to a buffer that is full. This is to make sure observability data is reliably processed in the order it was given. Additionally, as mentioned earlier, blocking induces backpressure and is an important signal to upstream components that they should slow down the acceptance or consumption of events. The result is that, while no data is lost, data piles up at the edge.
```

Comment:
```suggestion
Observability Pipelines destination's buffers are configured to block events, which means it waits indefinitely to write to a buffer that is full. This is to make sure observability data is reliably processed in the order it was given. Additionally, as mentioned earlier, blocking induces backpressure and signals upstream components to slow down event acceptance or consumption. As a result, although the system retains all data, it accumulates at the edge.
```
I was a little confused what this means "piles up at the edge" ? So just suggesting to reword again for clarity.

Reply:
@aliciascott edge is a technical terminology from edge computing:
https://vector.dev/docs/setup/going-to-prod/arch/agent/#processing-at-the-edge
https://www.techtarget.com/searchdatacenter/definition/edge-computing